Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -3395,17 +3396,26 @@ impl <T: ?Sized>Default for Object<T> {
     }
 }
 
-impl<T: ?Sized> Debug for Object<T> {
+impl<T: ?Sized + UpcastObject<dyn Any>> Debug for Object<T> {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
         self.fmt_print(f, false)
     }
 }
-impl <T: ?Sized> DafnyPrint for Object<T> {
+impl <T: ?Sized + UpcastObject<dyn Any>> DafnyPrint for Object<T> {
     fn fmt_print(&self, f: &mut Formatter<'_>, _in_seq: bool) -> std::fmt::Result {
-        write!(f, "<object>")
+        let option_string = UpcastObject::<dyn Any>::upcast(self.as_ref()).as_ref().downcast_ref::<String>().cloned();
```

Comment:
This clone is avoidable:

```suggestion
        let obj_any = UpcastObject::<dyn Any>::upcast(self.as_ref());
        let option_string = obj_any.as_ref().downcast_ref::<String>();
```